qol: Добавлено автонаведение на самую пострадавшую часть тела при использовании бинтов, китов и зашивалок

vovakons opened the pull request:
## Что этот ПР делает

1. При использовании бинта теперь не смотрит на то куда игрок нацелен, автоматически выбирает наиболее кровоточащую часть тела и накладывает бинт туда.
2. При использовании китов та же логика, но выбирается наиболее пострадавшая часть тела (по брут и берн урону, в зависимости от типа кита)
3. Зашивалка также автоматически найдет где зашивать.

## Почему это хорошо для игры

1. QoL
2. Понижаем градус духоты в меде после кровотеков.

## Тестирование

Локалка.

vovakons committed:
auto target for medicine stack items

Автонаводка по максимальному кровотечению для бинтов и зашивалок.
Автонаводка на часть тела по макс урону по типу для мазей и китов.

dageavtobusnick commented on code/game/objects/items/stacks/medical.dm:
```diff
@@ -111,8 +111,90 @@
+	if(!target_bodypart)
+		return
+
+	return target_bodypart.limb_zone
```
3 прока с практически одинаковым кодом. Don`t repeat yourself.

vovakons replied:
ну я хз как сделать один шаблонный прок, для этого надо бы лямбды фильтров передавать что ли по правильному, я просто хз как тут это реализовать

vovakons replied:
не хочу по флагу делать, будет выглядеть как костыль

dageavtobusnick replied:
> ну я хз как сделать один шаблонный прок, для этого надо бы лямбды фильтров передавать что ли по правильному, я просто хз как тут это реализовать

У нас такая штука как калбеки.

PlayerUnknown14 commented:
А может лучше оставить старую логику прицеливания для наборов, но перенаправлять на покоцанную конечность если выбранная — здорова?

vovakons committed:
use filter procs for remove code repeatings

Новый прок который получает прок-фильтр для выбора наиболее приоритетной
части тела.

dageavtobusnick commented on code/game/objects/items/stacks/medical.dm:
```diff
@@ -166,6 +167,64 @@
+	var/mob/living/carbon/human/human_target = target
+	var/obj/item/organ/external/target_bodypart = null
+	for(var/obj/item/organ/external/bodypart as anything in human_target.bodyparts)
+		var/accept = call(src, filter_proc)(arglist(list(current=bodypart, max=target_bodypart)))
```
```suggestion
		var/accept = call(src, filter_proc)(arglist(list(current = bodypart, max = target_bodypart)))
```

dageavtobusnick committed:
Update code/game/objects/items/stacks/medical.dm
